Add Obs-monitor website project code

EdwardSafford-NOAA opened the pull request:
Initial commit of website components including React, Vite, and tailwind related files.  These are all the files necessary to load into VS Code and generate the required website components.  The eventual (actual) website components are placed in a obsmon/dist/ directory which is intentionally **not** included in this PR.

This is, regrettably, a large PR of > 100 files.  Most of it won't make sense.  I _generally_ know what I'm working with now but only because I've been hacking away at this for the past month.  For review purposes I suggest focusing on the README  file and the src/*.jsx and src/components/*.jsx files.  The .jsx files are the React coded files which are fed into node.js to generate the user interface.  They contain html, javascript, and react coding elements.

I also note I've hit some sort of CI failure upon pushing this code to my fork so I'll have a look at that.  Yippee. 

Closes #63

Commit:
Ref #63

Initial commit of website components including React, Vite, and tailwind related files.

CoryMartin-NOAA commented on web/ObsMon/public/vite.svg:
Do we need this archived in the repo?

EdwardSafford-NOAA replied:
No, we shouldn't need this.  I'll 2x check the .gitignore file -- nothing in the public/ directory should be archived.

CoryMartin-NOAA commented on web/ObsMon/src/assets/react.svg:
Same question

CoryMartin-NOAA commented on web/ObsMon/src/components/OzoneObs.jsx:
```diff
@@ -0,0 +1,149 @@
+                ozoneSatellites.map(async (sat) => {
+                    const key = `${sat.satKey}_${sat.instrument}`;
+                    // const anomalyUrl = `/data/anomalyStatus_${sat.satKey}_${sat.instrument}_${cycleTime}.json`;
+                    const anomalyUrl = withBase(`data/anomalyStatus_${sat.satKey}_${sat.instrument}_${cycleTime}.json`);
```
Not sure if it should be in this PR or not, but I'm worried about a bunch of empty files with no anomalies slowing down and cluttering the filesystem. Can we have it assume no anomalies if the files are missing?

CoryMartin-NOAA commented on web/ObsMon/src/components/QObs.jsx:
```diff
@@ -0,0 +1,113 @@
+import React, { useEffect, useState, useMemo } from "react";
```
maybe I'm overcomplicating things but these .jsx files seem to be very similar. Could we use a template + python script to create them when the site is deployed?

EdwardSafford-NOAA replied:
I had similar thoughts though I was going to explore just combining them directly.  I think the input props could be used to differentiate the functionality.  I'd  like to explore that as a follow-on issue.

CoryMartin-NOAA reviewed (commented):
a few minor questions/comments. Also generally: a lot of files are missing the end of line character and show up with the "ghostbusters" symbol at the end on GitHub

Commit:
Ref #63

Remove empty anomalyStatus files.  Maybe correct missing eof markers?

Commit:
Ref #63

Attempt (again) to fix eof marker.  For the record I hate Windows.

Commit:
Ref #63

Once more unto the EOF breech....

EdwardSafford-NOAA commented on web/ObsMon/src/components/UvObs.jsx:
```diff
@@ -0,0 +1,114 @@
+
+        return enriched;
+    }, [assimilationStatus, anomalyStatus]);
+
```
For the record I think it's just _wild_ that we now have AI reviewing (mostly) AI code. Crazy world.  I'll look into this.

EdwardSafford-NOAA commented on web/ObsMon/src/components/UvObs.jsx:
```diff
@@ -0,0 +1,114 @@
+
+                    {/* Filtered Uv list */}
+                    {filteredUvTypes.map((uv) => (
+                        <TypeBlock
```
Valid point.  I blame ChatCPT.  :)

Commit:
Ref #63

Continue slaying the Windows EOF dragons.

Commit:
Ref #63

Last of the Windows EOF dragons?  I hope!

Commit:
Ref #63

Found one more EOF issue.

Commit:
Ref #63

Pull state setting out of useMemo callback and place in separate useEffect monitor per CoPilot's suggestion.

EdwardSafford-NOAA commented:
@CoryMartin-NOAA @kevindougherty-noaa I think I've gotten all the identified concerns corrected.  Please have another look when you have a moment.  I'll open a new issue to combine *Obs.jsx files.

CoryMartin-NOAA reviewed (approved):
Looks good, thanks @EdwardSafford-NOAA
